beacon-chain/blockchain/execution_engine.go

```diff
@@ -279,12 +280,20 @@ func (s *Service) pruneInvalidBlock(ctx context.Context, root, parentRoot, lvh [
 func (s *Service) getPayloadAttribute(ctx context.Context, st state.BeaconState, slot primitives.Slot, headRoot []byte) (bool, payloadattribute.Attributer) {
 	emptyAttri := payloadattribute.EmptyWithVersion(st.Version())
 
-	val, ok := s.trackedProposer(st, slot)
-	if !ok && !features.Get().PrepareAllPayloads {
-		return false, emptyAttri
+	// If it is an epoch boundary then process slots to get the right
+	// shuffling before checking if the proposer is tracked. Otherwise
+	// perform this check before. This is cheap as the NSC has already been updated.
+	var val cache.TrackedValidator
+	var ok bool
+	e := slots.ToEpoch(slot)
+	stateEpoch := slots.ToEpoch(st.Slot())
+	if e == stateEpoch {
+		val, ok = s.trackedProposer(st, slot)
+		if !ok && !features.Get().PrepareAllPayloads {
+			log.Debug("returning from getPayloadAttribute because validator is not tracked")
```

Comment:
This log is emitted for each block, which seems excessive. While its purpose is to track validators, it currently extends to all nodes, even those not running any validators, resulting in unnecessary logging.